Python/pylifecycle.c

```diff
@@ -2145,8 +2145,11 @@ wait_for_thread_shutdown(void)
     PyObject *result;
     PyObject *threading = _PyImport_GetModuleId(&PyId_threading);
     if (threading == NULL) {
-        /* threading not imported */
-        PyErr_Clear();
+        if (PyErr_Occurred()) {
+            /* unrecoverable error */
+            PyErr_Clear();
```

Comment:
Should not we call `PyErr_WriteUnraisable()`?

Reply:
Maybe, I updated the PR now because it's used in other places as well.

But `PyErr_WriteUnraisable()` contains a lot of code that could fail again; I'd be more comfortable in general with `PyErr_Clear()` and a simple `fprintf()`.

On the other hand, in this particular case (thread finalization) stack exhaustion shouldn't be a problem.